x-pack/plugin/logsdb/src/main/java/org/elasticsearch/xpack/logsdb/LogsdbIndexModeSettingsProvider.java

```diff
@@ -110,14 +118,56 @@ && matchesLogsPattern(dataStreamName)) {
                 settingsBuilder.put(SourceFieldMapper.INDEX_MAPPER_SOURCE_MODE_SETTING.getKey(), SourceFieldMapper.Mode.STORED.toString());
             }
         }
+
+        // Inject routing path matching sort fields.
+        if ((isLogsDB || resolveIndexMode(settings.get(IndexSettings.MODE.getKey())) == IndexMode.LOGSDB)
```

Comment:
```suggestion
        if ((isLogsDB)
```

This should also work? If index.mode is specified than `isLogsDB` should always be true?

Reply:
I think this is needed, in case `templateIndexMode` is not set to LOGSDB but `IndexSettings.MODE` is set to LOGSDB.

Reply:
I don't think this can happen? The `templateIndexMode` is sources from either index settings or template setting in `MetaCreateIndexService#aggregateIndexSettings(...)` (line 1007), so I think just checking `isLogsDB` variable is ok?